commit cfabc1bbee547e74b70cbfb0bd66b30bbdc9d8e5
Author: Stéphane Graber <stgraber@ubuntu.com>
Date:   Sat Jul 1 01:33:31 2017 -0400

    config: Try to be clever about ":" in snapshots
    
    If we detect a ":" in the provided string and can't find any remote with
    that name and we detect a "/" in the resource name, then assume we're
    dealing with a snapshot containing ":" and select the default remote.
    
    Launchpad: https://bugs.launchpad.net/ubuntu/+source/lxd/+bug/1694855
    Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

diff --git a/lxc/config/remote.go b/lxc/config/remote.go
index 3317ccc..e5d255e 100644
--- a/lxc/config/remote.go
+++ b/lxc/config/remote.go
@@ -21,11 +21,16 @@ type Remote struct {
 func (c *Config) ParseRemote(raw string) (string, string, error) {
 	result := strings.SplitN(raw, ":", 2)
 	if len(result) == 1 {
-		return c.DefaultRemote, result[0], nil
+		return c.DefaultRemote, raw, nil
 	}
 
 	_, ok := c.Remotes[result[0]]
 	if !ok {
+		// Attempt to play nice with snapshots containing ":"
+		if shared.IsSnapshot(raw) && strings.Contains(result[0], "/") {
+			return c.DefaultRemote, raw, nil
+		}
+
 		return "", "", fmt.Errorf("The remote \"%s\" doesn't exist", result[0])
 	}
 
